src/btree/bt_curprev.c

```diff
@@ -723,6 +703,20 @@ __wt_btcur_prev(WT_CURSOR_BTREE *cbt, bool truncating)
     F_CLR(cursor, WT_CURSTD_KEY_SET | WT_CURSTD_VALUE_SET);
 
     WT_ERR(__wt_cursor_func_init(cbt, false));
+
+    /*
+     * If we have a bound set we should position our cursor appropriately if it isn't already
+     * positioned. In one scenario this function returns and needs to walk to the next record. In
+     * that case we set a boolean.
+     */
+    if (F_ISSET(cursor, WT_CURSTD_BOUND_UPPER) && !WT_CURSOR_IS_POSITIONED(cbt)) {
+        WT_ERR(__wt_btcur_bounds_row_position(session, cbt, false, &need_walk));
+        if (!need_walk) {
+            __wt_value_return(cbt, cbt->upd_value);
+            return (0);
```

Comment:
Is it safe to skip all the code after the err: label in this case? It's doing a lot of work, and the WT_ERR above makes me wonder whether it's correct to skip that logic. i.e: should this be a `goto err;` or `goto done;` instead of a `return (0)`?

Reply:
Good point, I have created a done label, and used goto here.